src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java

```diff
@@ -43,26 +56,43 @@ private static StatsdReporter provideStatsDHandlers() {
         return provider.get();
     }
 
-    private static void runAfterSocketInit(Runnable runnable) {
-        Executors.newScheduledThreadPool(1).schedule(runnable, 100, TimeUnit.MILLISECONDS);
-    }
-
-
-    @BeforeClass
-    public static void startMockStatsDServer() {
+    public void startMockStatsDServer() throws InterruptedException, ExecutionException, TimeoutException {
+        CountDownLatch startupCountDownLatch = new CountDownLatch(2);
+        AtomicBoolean canListen = new AtomicBoolean();
         StatsdReporter handlers = provideStatsDHandlers();
-        IntegrationUdpServer udpServer = provideNewServer(metricsConfig.getStatsdPort());
-        IntegrationUdpServer udpServerDD = provideNewServer(metricsConfig.getDogstatsdPort());
-        runAfterSocketInit(() -> {
+        ExecutorService executorService = Executors.newFixedThreadPool(3);
+        IntegrationUdpServer udpServer = provideNewServer(metricsConfig.getStatsdPort(), startupCountDownLatch, canListen, executorService);
+        IntegrationUdpServer udpServerDD = provideNewServer(metricsConfig.getDogstatsdPort(), startupCountDownLatch, canListen, executorService);
+        CompletableFuture<String> sender = CompletableFuture.supplyAsync(() -> {
+            try {
+                startupCountDownLatch.await(15000, TimeUnit.MILLISECONDS);
+                log.info("yay");
+                log.info(Long.valueOf(startupCountDownLatch.getCount()).toString());
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+                log.error(e.getMessage());
+                fail();
+            }
+            log.info("gonna count now");
             handlers.count("countTest", 1, tags);
             handlers.increment("closeSocket", tags);
-        });
+            log.info("done with handlers");
+            canListen.set(true);
+            return "done";
+        }, executorService);
+        log.info("about to get futures");
+        CompletableFuture.allOf(sender, udpServer.getServerFuture(), udpServerDD.getServerFuture())
+                .get(15000, TimeUnit.MILLISECONDS);
```

Comment:
I see one more 15000. Can you move this to a constant variable?

Reply:
Technically the two values aren't related, so I'm not sure changing this to a constant makes sense.